File: Dataset/Leetcode/train/5/239.py
 class Solution(object):
    def XXX(self, s):
        length = len(s)
        if length < 2:
            return '' if length==0 else s
        dp = [[0] * length for i in range(length)]
        start = 0
        end = 0
        #Base Case 长度为1和长度为2
        for i in range(length-1):
            dp[i][i] = 1
            if s[i]==s[i+1]:
                dp[i][i+1] = 1
                start = i
                end = i+1
        # 枚举子串的长度
        for l in range(3,length+1):
            low = 0
            while low+l-1 < length:
                high = low + l -1
                if s[low] == s[high] and dp[low+1][high-1]:
                    dp[low][high] = 1
                    start = low
                    end = high
                low+=1
        return s[start:end+1]

File: Dataset/Leetcode/train/9/239.cpp
class Solution {
public:
    bool XXX(int x) {
        //一位一位地反转给定数，当反转数大于或等于给定数字时停止
        //停止时有三种可能：
        //1.最后一位数字为0，前面的数字构成反转数
        //2.给定数有奇数位数（反转数与给定数字有一个数量级的差距，需要消除这个数量级）
        //3.给定数有偶数位数
        if((x < 0) || (x % 10 == 0 && x != 0))
            return false;

        int reverse = 0;
        while(reverse < x)
        {
            reverse = reverse * 10 + x % 10;
            x = x / 10;
        }
        return reverse == x || reverse / 10 == x;
    }
};

